api/client/cluster/client.go

```diff
@@ -192,3 +193,68 @@ func (c *clusterClient) EraseAlert(resource api.ResourceType, alertID int64) err
 	}
 	return nil
 }
+
+func (c *clusterClient) SetClusterSecretKey(secretKey string, override bool) error {
+	request := c.c.Post().Resource("/setclustersecretkey")
+	request.QueryOption(secrets.ClusterSecretKey, secretKey)
+
+	stOverride := "false"
+	if override {
+		stOverride = "true"
+	}
+	request.QueryOption(secrets.OverrideSecrets, stOverride)
+	resp := request.Do()
+	if resp.Error() != nil {
+		return resp.FormatError()
+	}
+	return nil
+}
+
+func (c *clusterClient) SetSecret(secretid string, secretvalue string) error {
+	request := c.c.Post().Resource("/setsecret")
+	request.QueryOption(secrets.SecretKey, secretid)
```

Comment:
create a secretrequest and secret response struct
marshal the data in the struct instance
and send the instance in the Body

Reply:
I did but internal PX secret implemenation is such way that it accept  secretid as params and data as body.But will ask aditya about same